### ISSUE TITLE
Distance matrix as the only input for factoextra package

### ISSUE THREAD

**Issue by SimonPoirier** [kassambara/factoextra#105]
Hi!

I would like to know how would it be possible to use a distance matrix as the only input for the different functions of factoextra package such as: fviz_nbclust, fviz_silhouette etc...
Indeed, considering my data, I can not have a classical matrix with samples and descriptors.

Thanks a lot for your answer.

Best regards 

Simon

**Comment by kassambara** [comment]
Yes — most clustering entry points accept a precomputed dissimilarity (`"dist"`) object, so you don't need a samples×features matrix:

```r
library(factoextra)
d <- my_dist            # an object of class "dist"
# choose k
fviz_nbclust(as.matrix(d), cluster::pam, method = "silhouette", diss = d)
# cluster
res <- eclust(d, "hclust", k = 4)     # or eclust(d, "pam", k = 4); or hcut(d, k = 4)
fviz_dend(res)
# silhouette from any clustering computed on the dist
fviz_silhouette(cluster::silhouette(res$cluster, d))
```

Note: `fviz_nbclust(method = "gap_stat")` still needs the original data (the gap statistic simulates reference data), so it can't run from a dissimilarity alone. Closing as answered.
